src/observable/fromArrayLike.ts

```diff
@@ -0,0 +1,66 @@
+import {Scheduler} from '../Scheduler';
+import {Observable} from '../Observable';
+import {Subscriber} from '../Subscriber';
+import {Subscription} from '../Subscription';
+
+export class ArrayLikeObservable<T> extends Observable<T> {
+
+  private mapFn: (x: number, y: any) => T;
+
+  static create<T>(arrayLike: ArrayLike<T>, mapFn: (x: number, y: any) => T, thisArg: any, scheduler?: Scheduler) {
+    return new ArrayLikeObservable(arrayLike, mapFn, thisArg, scheduler);
+  }
+
+  static dispatch(state: any) {
+
+    const { arrayLike, index, count, mapFn, subscriber } = state;
+
+    if (index >= count) {
+      subscriber.complete();
+      return;
+    }
+
+    const result = mapFn ? mapFn(index, arrayLike[index]) : arrayLike[index];
+    subscriber.next(result);
+
+    if (subscriber.isUnsubscribed) {
+      return;
+    }
+
+    state.index = index + 1;
+
+    (<any> this).schedule(state);
+  }
+
+  // value used if Array has one value and _isScalar
+  value: any;
+
+  constructor(public arrayLike: ArrayLike<T>, mapFn: (x: number, y: any) => T, thisArg: any, public scheduler?: Scheduler) {
```

Comment:
`mapFn`'s arguments are backwards, I think. It should be: `(item: any, index: number) => T`.